core/logger/logger.go

```diff
@@ -191,10 +191,18 @@ func NewLogger() (Logger, func() error) {
 
 	var fileMaxSize utils.FileSize
 	fileMaxSize, invalid = envvar.LogFileMaxSize.Parse()
-	c.FileMaxSize = int(fileMaxSize)
 	if invalid != "" {
 		parseErrs = append(parseErrs, invalid)
 	}
+	if fileMaxSize == 0 || invalid != "" {
+		c.FileMaxSize = 0 // default is 100Mb
+		parseErrs = append(parseErrs, "LogFileMaxSize will default to 100Mb")
```

Comment:
It seems brittle to depend on this default being set from somewhere else. Could we just set it ourselves here explicitly?
```suggestion
		c.FileMaxSize = 100
		parseErrs = append(parseErrs, "LogFileMaxSize will default to 100Mb")
```

Reply:
Fixed